include/bbp/sonata/output.h

```diff
@@ -0,0 +1,143 @@
+#pragma once
+
+#include <algorithm>
+#include <map>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <highfive/H5File.hpp>
+
+#include <bbp/sonata/population.h>
+
+namespace H5 = HighFive;
+
+namespace bbp {
+namespace sonata {
+
+// KeyType will be NodeID for somas report and pair<NodeID, uint32_t> for elemets report
+template <typename KeyType>
+struct SONATA_API DataFrame {
+    using DataType = std::map<KeyType, std::vector<float>>;
+    std::vector<double> index;
+    DataType data;
+};
+
+/**
+  const SpikeReader file(filename);
+  auto pops = file.getPopulationNames();
+  for (const auto& data: file[pops[0]].get(Selection{12UL, 34UL, 58UL})) {
+      NodeID node_id;
+      double timestamp;
+      std::tie(node_id, timestamp) = data;
+      std::cout << "[" << timestamp << "] " << node_id << std::endl;
+  }
+*/
+class SONATA_API SpikeReader
+{
+  public:
+    class Population
+    {
+      public:
+        enum class Sorting : char {
+            none = 0,
+            by_id = 1,
+            by_time = 2,
+        };
+
+        using Spike = std::pair<NodeID, double>;
+        using Spikes = std::vector<Spike>;
+        Spikes get(const Selection& node_ids = Selection({}),
+                   double tstart = -1,
+                   double tstop = -1) const;
+        Sorting getSorting() const;
+
+      private:
+        Population(const std::string& filename, const std::string& populationName);
+
+        Spikes spikes_;
+        Sorting sorting_ = Sorting::none;
+        // Use for clamping of user values
+        double tstart_, tstop_;
```

Comment:
I think that being in the report reader context, this doesn't need to be emphasized, otherwise almost everything should be prefixed with report_

Reply:
I see your point, I'm just reading the code, and `tstart`/`tend` is used for the desired times of interest, as well as maximum report time, so I thought it might be worth differentiating them.